src/Raven.Server/RavenServerStartup.cs

```diff
@@ -258,6 +262,12 @@ await using (var writer = new AsyncBlittableJsonTextWriter(ctx, context.Response
             }
         }
 
+        private static void CheckDatabaseShutdownAndThrowIfNeeded(DocumentDatabase database, ref Exception e)
+        {
+            if(database != null && database.DatabaseShutdown.IsCancellationRequested && e is TaskCanceledException)
```

Comment:
1. Shouldn't we check for OperationCanceledException?
2. Would prefer to check for ex type first, then for other items

Reply:
Done